src/crypto/tls/common.go

```diff
@@ -674,6 +710,23 @@ type Config struct {
 	// used for debugging.
 	KeyLogWriter io.Writer
 
+	// ClientECHConfigs are the parameters used by the client when it offers the
+	// ECH extension. If a suitable configuration is found and the client
+	// supports TLS 1.3, then ECH will be offered in this handshake.
+	ClientECHConfigs []ECHConfig
+
+	// ClientECHCovertext enables generation of covertext for the ECH extension.
+	// If set, and the client does not offer a real ECH extension, then it will
+	// generate a dummy ECH extension.
+	ClientECHCovertext bool
```

Comment:
The spec does not use the term "covertext", but "GREASE". I would avoid using non-standard terminology.

Additionally, I'd suggest replacing this member by a `ClientECHEnabled` or `ECHEnabled` bool. The GREASE functionality is an protocol detail that the application developer should not be concerned about.

If disabled, then ClientECHConfigs would be ignored.
If enabled, and `ClientECHConfigs` is empty, then a GREASE value will be used.

If enabled, and `ClientECHConfigs` is non-empty, then ECH will be enabled as usual. If an unsupported config is encountered... I suppose the handshake could fail just to be sure.

Reply:
> The spec does not use the term "covertext", but "GREASE". I would avoid using non-standard terminology.

I don't think "GREASE" is correctly specified, as of yet. In particular, the spec refers to the existing RFC for GREASE, which designates particular code points for extension greasing. "Covertext" is a much better word.

> Additionally, I'd suggest replacing this member by a `ClientECHEnabled` or `ECHEnabled` bool. The GREASE functionality is an protocol detail that the application developer should not be concerned about.

Done.